content/en/agent/kubernetes/daemonset_setup.md

```diff
@@ -201,39 +201,68 @@ To enable [Log collection][8] with your DaemonSet:
       (...)
     ```
 
-Learn more about this in [the Docker log collection documentation][9].
+Learn more about this in [the Docker log collection documentation][10].
 
-### Trace Collection
+### APM and Distributed Tracing
 
-To enable trace collection with your DaemonSet:
+To enable APM by allowing incoming data from port 8126, set the `DD_APM_NON_LOCAL_TRAFFIC` variable to true in your *env* section:
 
-1. Set the Node IP and port as environment variables for your application containers:
+```
+(...)
+      env:
+        (...)
+        - name: DD_APM_NON_LOCAL_TRAFFIC
+          value: "true"
+(...)
+```
 
-    ```
-    apiVersion: apps/v1
-    kind: Deployment
-    ...
-        spec:
-          containers:
-          - name: <CONTAINER_NAME>
-            image: <CONTAINER_IMAGE>/<TAG>
-            env:
-              - name: DD_AGENT_HOST
-                valueFrom:
-                  fieldRef:
-                    fieldPath: status.hostIP
-    ```
+Then, forward the port of the Agent to the host. 
+
+```
+(...)
+      ports:
+        (...)
+        - containerPort: 8126
+          hostPort: 8126
+          name: traceport
+          protocol: TCP
+(...)
+```
 
-2. Uncomment the `# hostPort: 8126` line in your `datadog-agent.yaml` manifest:
-  This exposes the Datadog Agent tracing port on each of your Kubernetes nodes.
+Use the downward API to pull the host IP; the application container needs an environment variable that points to `status.hostIP`. The Datadog container Agent expects this to be named `DD_AGENT_HOST`:
+
+```
+apiVersion: apps/v1
+kind: Deployment
+...
+    spec:
+      containers:
+      - name: <CONTAINER_NAME>
+        image: <CONTAINER_IMAGE>/<TAG>
+        env:
+          - name: DD_AGENT_HOST
+            valueFrom:
+              fieldRef:
+                fieldPath: status.hostIP
+```
+
+Finally, point your application-level tracers to where the Datadog Agent host is using the environment variable `DD_AGENT_HOST`. For example, in Python:
+
+```
+import os
+from ddtrace import tracer
+
+tracer.configure(
+    hostname=os.environ['DD_AGENT_HOST'],
+    port=os.environ['DD_TRACE_AGENT_PORT'],
+)
+```
 
-    **Warning**: The `hostPort` parameter opens a port on your host. Make sure your firewall only allows access from your applications or trusted sources. 
```

Comment:
Should we remove the warning? if so, should we remove it from the DogStatsD section below as well?